starboard/android/shared/drm_system.cc

```diff
@@ -254,112 +234,38 @@ DrmSystem::~DrmSystem() {
   }
 }
 
-DrmSystem::SessionUpdateRequest::SessionUpdateRequest(
-    int ticket,
-    const char* type,
-    const void* initialization_data,
-    int initialization_data_size) {
-  JniEnvExt* env = JniEnvExt::Get();
-  j_ticket_ = static_cast<jint>(ticket);
-  j_init_data_ =
-      ByteArrayFromRaw(initialization_data, initialization_data_size);
-  j_mime_ = env->NewStringStandardUTFOrAbort(type);
-}
-
-void DrmSystem::SessionUpdateRequest::ConvertLocalRefToGlobalRef() {
-  if (!references_are_global_) {
-    JniEnvExt* env = JniEnvExt::Get();
-    j_init_data_ = env->ConvertLocalRefToGlobalRef(j_init_data_);
-    j_mime_ = env->ConvertLocalRefToGlobalRef(j_mime_);
-    references_are_global_ = true;
-  }
-}
-
-DrmSystem::SessionUpdateRequest::~SessionUpdateRequest() {
-  JniEnvExt* env = JniEnvExt::Get();
-  if (references_are_global_) {
-    env->DeleteGlobalRef(j_init_data_);
-    env->DeleteGlobalRef(j_mime_);
-  } else {
-    env->DeleteLocalRef(j_init_data_);
-    env->DeleteLocalRef(j_mime_);
-  }
-  j_init_data_ = nullptr;
-  j_mime_ = nullptr;
-}
-
-void DrmSystem::SessionUpdateRequest::Generate(
-    jobject j_media_drm_bridge) const {
-  JniEnvExt* env = JniEnvExt::Get();
-  env->CallVoidMethodOrAbort(j_media_drm_bridge, "createSession",
-                             "(I[BLjava/lang/String;)V", j_ticket_,
-                             j_init_data_, j_mime_);
-}
-
 void DrmSystem::GenerateSessionUpdateRequest(int ticket,
                                              const char* type,
                                              const void* initialization_data,
                                              int initialization_data_size) {
-  std::unique_ptr<SessionUpdateRequest> session_update_request(
-      new SessionUpdateRequest(ticket, type, initialization_data,
-                               initialization_data_size));
-  if (created_media_crypto_session_.load()) {
-    session_update_request->Generate(j_media_drm_bridge_);
-  } else {
-    // Defer generating the update request.
-    session_update_request->ConvertLocalRefToGlobalRef();
-    ScopedLock scoped_lock(mutex_);
-    deferred_session_update_requests_.push_back(
-        std::move(session_update_request));
-  }
-  // |update_request_callback_| will be called by Java calling into
-  // |onSessionMessage|.
+  const uint8_t* initialization_data_ptr =
+      static_cast<const uint8_t*>(initialization_data);
```

Comment:
nit: if we use jbyte here, we can save the cast in `ByteArrayFromRaw()`.  Also, consider add another function (e.g. `ByteVectorFromRaw()`) to convert a raw buffer to std::vector<jbyte>.